{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data-Independent Mass Spectrometry: From RAW files to Analyses\n",
    "\n",
    "In this notebook, I'll outline the workflow necessary to take my *Crassostrea gigas* Data-Independent Mass Spectrometry (DIA) data from `RAW` files off of the mass spectrometer to a `.csv` usable in R Studio. DIA allows us to identify all peptides in each oyster sample without having any prior knowledge of what we may find.\n",
    "\n",
    "*Note: This pipeline relies heavily on Windows-based programs*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Collect Materials"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Software Dependencies**:\n",
    "\n",
    "- [Skyline daily](https://skyline.ms/project/home/software/Skyline/daily/register-form/begin.view?)\n",
    "  - This version of Skyline will udpates as the software is modified. You need permission to download it. The version of Skyline daily used in this notebook is Skyline-daily (64-bit) 3.7.1.11446 (as of 2017-10-11). \n",
    "- [MSConvert]()\n",
    "  - Special version of MSConvert modified by Austin in Genome Sciences on 2017-04-18.\n",
    "- [R](https://www.r-project.org) and [R Studio](https://www.rstudio.com)\n",
    "  - R version 3.4.0 (2017-04-21) -- \"You Stupid Darkness\"\n",
    "  - R Studio Version 1.0.143\n",
    "  \n",
    "**Specific files (in order used)**:\n",
    "\n",
    "- [config file for MSConvert](https://github.com/RobertsLab/project-oyster-oa/blob/master/analyses/DNR_MSConvert_20170412/config_fix_20170413.txt)\n",
    "  - This file contains all of the information MSConvert needs to demultiplex RAW files. It includes an arugment specific to the isolation scheme used by the mass spectrometer to collect data.\n",
    "- [.blib file from `pecanpie`](http://owl.fish.washington.edu/spartina/DNR_Skyline_20170524/2017-05-23-oyster-desearleinated.blib)\n",
    "  - A .blib file is a library of all peptides you want to identify in your samples. For this experiment, the .blib contains the entire *C. gigas* proteome. This is the same .blib file used in my DIA analyses. This file was created by Emma Timmins-Schiffman and her team in `brecan`, a version of `pecanpie` produced by Brian Searle in Genome Sciences. Jarrett Egertson wrote a script to get rid of incompatibilities generated by `brecan`. The final compatible document is found above. For more information regarding .blib creation, see this [lab notebook entry](https://yaaminiv.github.io/Skyline-Attempt-3/).\n",
    "- [Undigested **FASTA** background proteome, including QC protein sequence](http://owl.fish.washington.edu/spartina/DNR_Skyline_20170427/Combined-gigas-QC.fasta)\n",
    "- [RAW files](http://owl.fish.washington.edu/spartina/January_2017_DNR_Raw_Data/Oyster_raw_files/)\n",
    "- [Key for samples](http://owl.fish.washington.edu/spartina/January_2017_DNR_Raw_Data/2017_January_23.csv)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Demultiplex RAW data in MSConvert\n",
    "\n",
    "DIA mass spectrometry uses overlapping mass-charge (m/z) windows to capture all peptides in a sample. To discern which peptides and transitions were found in each window, RAW files are demultiplexed. This allows us to parse out peptide abundance data from each individual window. Demultiplexing requires use of the MSConvert command line interface."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 2a: Install the MSConvert CLI\n",
    "\n",
    "Unzip the file in Step 1 in your desired working directory. This should be the same directory with the RAW files to convert."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 2b: Convert RAW files\n",
    "\n",
    "Use the following code to demultiplex all RAW files and convert them to .mzML files:\n",
    "\n",
    "`msconvert.exe -c config_fix_20170413.txt *.raw`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Step 3: Create a new Skyline Document"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3a: Import the Spectral Library\n",
    "\n",
    "1. Open a new Skyline file (\"Blank Document\")\n",
    "2. Under Settings >> Peptide Settings >> Library, click \"Edit List.\" The *C. gigas* .blib should already be on the list (2017-05-23-oyster-desearlinated).\n",
    "3. If the .blib is not already on the list, click \"Add\"\n",
    " - Name the library and select the .blib file\n",
    " - After clicking \"OK,\" select the correct library from the list\n",
    "4. Under pick peptides matching, select \"Library\"\n",
    "\n",
    "<img width=\"271\" alt=\"screen shot 2017-09-13 at 2 01 34 am\" src=\"https://user-images.githubusercontent.com/22335838/31026852-c1339b8a-a4fc-11e7-8a22-37a549a91524.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3b: Add Background Proteome\n",
    "\n",
    "1. Settings >> Peptide Settings >> Digestion\n",
    "2. Select \"Gigas-4-27\" under \"Background proteome\"\n",
    "3. If \"Gigas-4-27\" is not an option\n",
    "  - Select \"Add\" under \"Background proteome\"\n",
    "  - Name background\n",
    "  - Click \"Create\" under \"Proteome file\" to choose where to save the background\n",
    "  - Click \"Add file\" under \"FASTA files\", and select your background proteome fasta file\n",
    "  - After the FASTA loads, Skyline may prompt you about deleting repeated protein sequences. Select \"OK\"\n",
    "  \n",
    "    ![unnamed-1](https://user-images.githubusercontent.com/22335838/31412862-93c9d204-adcb-11e7-9fd4-68eca9957ee7.png)  \n",
    "\n",
    "4. Select \"Trypsin [KR | P]\" under \"Enzyme\n",
    "\n",
    "<img width=\"270\" alt=\"screen shot 2017-09-13 at 2 01 25 am\" src=\"https://user-images.githubusercontent.com/22335838/31027061-b5cd9c0e-a4fd-11e7-94a0-291e42fb8169.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3c: Adjust Peptide Settings\n",
    "\n",
    "1. Under the Prediction tab, make sure \"none\" is selected for retention time predictor\n",
    "\n",
    "    <img width=\"271\" alt=\"screen shot 2017-09-13 at 2 01 30 am\" src=\"https://user-images.githubusercontent.com/22335838/31027116-e58bdb36-a4fd-11e7-94bf-c3551aee8381.png\">\n",
    "\n",
    "2. Under the Filter tab, click \"Auto-select all matching peptides\" and ensure 2 and 25 are the Min and Max lengths. Under \"Exclude N-Terminal AAs\" enter zero.\n",
    "\n",
    "    <img width=\"268\" alt=\"screen shot 2017-09-13 at 2 01 32 am\" src=\"https://user-images.githubusercontent.com/22335838/31027164-11eca5d4-a4fe-11e7-8064-30a50f6f150b.png\">\n",
    "\n",
    "3. Under the Modification tab, select \"Carbamidomethyl (C)\" under structural modifications, \"heavy\" Isotope label type, and \"light\" Internal standard type.\n",
    "\n",
    "    <img width=\"271\" alt=\"screen shot 2017-09-13 at 2 01 36 am\" src=\"https://user-images.githubusercontent.com/22335838/31027224-55f0238c-a4fe-11e7-8756-6a8139026253.png\">\n",
    "\n",
    "4. Use the following settings under the Quantification tab\n",
    "\n",
    "    <img width=\"273\" alt=\"screen shot 2017-09-13 at 2 01 37 am\" src=\"https://user-images.githubusercontent.com/22335838/31027278-82438f64-a4fe-11e7-9683-a57c386f2243.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3d: Populate Analyte Tree\n",
    "\n",
    "1. Under File >> Import, select \"Transition List\"\n",
    "2. Select the transition list (.csv) outlined in Step 1. This will populate the analyte tree only with targeted proteins and quality control (PRTC) peptides.\n",
    "\n",
    "Skyline will keep the proteins, peptides and transitions that match what it finds in the library provided in Step 2a."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3e: Adjust Transition Settings\n",
    "\n",
    "1. **Settings >> Transition Settings >> Prediction**\n",
    " - Precursor mass: Monoisotopic\n",
    " - Product ion mass: Monoisotopic\n",
    " - Collision energy: Thermo TSQ Vantage\n",
    " - Declustering potential: None\n",
    " - Optimization library: None\n",
    " - Compensation voltage: None\n",
    " - DO NOT select \"Use optimization values when present\"\n",
    " \n",
    " <img width=\"273\" alt=\"screen shot 2017-09-13 at 2 01 52 am\" src=\"https://user-images.githubusercontent.com/22335838/31027664-d9614fce-a4ff-11e7-9e83-48d95b8d34a0.png\">\n",
    "\n",
    "2. **Settings >> Transition Settings >> Filter >> Peptides**\n",
    " - Precursor charges: 2, 3\n",
    " - Ion charges: 1, 2\n",
    " - Ion types: y\n",
    " - Product ion selection\n",
    "   - From: ion 2\n",
    "   - To: last ion\n",
    "   - DO NOT select any Special ions\n",
    " - DO NOT specify any Precursor m/z exclusion window\n",
    " - DO select \"Auto-select all matching transitions\"\n",
    " \n",
    " <img width=\"271\" alt=\"screen shot 2017-09-13 at 2 01 53 am\" src=\"https://user-images.githubusercontent.com/22335838/31027761-4823f6e6-a500-11e7-9504-3a6fba2c6894.png\">\n",
    " \n",
    "3. **Transition Settings >> Library**\n",
    " - Ion match tolerance: 0.5 m/z\n",
    " - DO NOT select \"If a library spectrum is available, pick its most intense ions\"\n",
    " \n",
    " <img width=\"275\" alt=\"screen shot 2017-09-13 at 2 01 55 am\" src=\"https://user-images.githubusercontent.com/22335838/31027790-708c8a76-a500-11e7-954f-b39b5c70ac12.png\">\n",
    " \n",
    "4. **Transition Settings >> Instrument**\n",
    " - Min m/z: 100\n",
    " - Max m/z: 2000\n",
    " - DO NOT select \"Dynamic min product m/z\"\n",
    " - Method match tolerance m/z: 0.055 m/z\n",
    " - DO NOT specify any other settings on this tab\n",
    " \n",
    " <img width=\"272\" alt=\"screen shot 2017-09-13 at 2 01 57 am\" src=\"https://user-images.githubusercontent.com/22335838/31027851-a62f48ee-a500-11e7-8f99-bb735a401e82.png\">\n",
    " \n",
    "5. **Transition Settings >> Full-Scan**\n",
    "   - Isotope peaks included: Count\n",
    "   - Precursor mass analyzer: Orbitrap\n",
    "   - Peaks: 3\n",
    "   - Resolving power: 60,000\n",
    "   - At: 400 m/z\n",
    "   - Isotope labeling enrichment: Default\n",
    "   - Acquisition method: Targeted\n",
    "   - Product mass analyzer: Centroided\n",
    "   - Isolation scheme: None\n",
    "   - Mass Accuracy: 20 ppm\n",
    "   - DO NOT select \"Use high-sensitivity extraction\"\n",
    "   - Select \"Use only scans within 2 minutes of MS/MS IDs\"\n",
    " \n",
    " <img width=\"271\" alt=\"screen shot 2017-09-13 at 2 01 58 am\" src=\"https://user-images.githubusercontent.com/22335838/31028121-a3242c90-a501-11e7-9443-d0896e6d7dd6.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 5: Clean Data\n",
    "\n",
    "In this step, files without any data will be removed and peaks in Skyline will be verfied against predicted retention times."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 5a: Check that all QC peptides are chosen correctly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 5b: Spot check peptides"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 5c: Export data\n",
    "\n",
    "To proceed with downstream analyses, data must be exported from Skyline as a .csv."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Under File > Export > Report, use the following settings to export Skyline data as a .csv. However, do not include the \"Total Ion Current Area\" option.\n",
    "\n",
    "![30132381-03f87a94-9305-11e7-8dfa-812e738abbd0](https://user-images.githubusercontent.com/22335838/30983237-ad0f3056-a43e-11e7-8d51-d99207d262e1.png)\n",
    "\n",
    "Exported data can be found [here](http://owl.fish.washington.edu/spartina/DNR_SRM_20170728/Analyses/2017-09-12-Gigas-SRM-ReplicatesOnly-PostDilutionCurve-NoPivot-RevisedSettings-Report.csv)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 5d: Normalize peak areas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
